### PR TITLE
navidrome: fix darwin build, version numbering, and tests

### PR DESCRIPTION
**Pull request by vs49688**
###### Description of changes

Fix darwin build, version numbering, and tests.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vs49688** [comment]
@ofborg build navidrome

File: pkgs/servers/misc/navidrome/default.nix [hunk 1]
```diff
@@ -52,7 +58,9 @@ buildGoModule {
       --prefix PATH : ${lib.makeBinPath [ ffmpeg ]}
   '';
 
-  doCheck = false;
+  preCheck = ''
+    export GOFLAGS=''${GOFLAGS//-trimpath/}
```
Comment (Mic92) [inline pkgs/servers/misc/navidrome/default.nix:62]: Can you explain what this line fixes?

Comment (vs49688) [reply pkgs/servers/misc/navidrome/default.nix:62]: When using `-trimpath`, the unit tests can't find any of their needed files, so they fail.

Comment (Mic92) [reply pkgs/servers/misc/navidrome/default.nix:62]: We use trimpath to remove references to the go compiler in the resulting build: 

https://github.com/NixOS/nixpkgs/blob/2f0c3be57c348f4cfd8820f2d189e29a685d9c41/pkgs/development/go-modules/generic/default.nix#L145

Comment (Mic92) [reply pkgs/servers/misc/navidrome/default.nix:62]: But for tests, there should be no harm in having those references.

Comment (Mic92) [reply pkgs/servers/misc/navidrome/default.nix:62]: cc @zowoq

Comment (Mic92) [reply pkgs/servers/misc/navidrome/default.nix:62]: https://github.com/NixOS/nixpkgs/pull/182665

Comment (zowoq) [reply pkgs/servers/misc/navidrome/default.nix:62]: > But for tests, there should be no harm in having those references.

Yes, should be fine for tests as we also have `disallowedReferences` for `go`.

File: pkgs/servers/misc/navidrome/default.nix [hunk 2]
```diff
@@ -41,7 +41,13 @@ buildGoModule {
 
   buildInputs = [ taglib zlib ];
 
-  CGO_CFLAGS = [ "-Wno-return-local-addr" ];
+  ldflags = [
+    "-s" "-w"
```
Comment (squalus) [inline pkgs/servers/misc/navidrome/default.nix:45]: Why is `-s -w` desired here? My understanding is that this strips some stuff out of the binary. It looks like some things do this in nixpkgs, but most go builds do not.

Comment (vs49688) [reply pkgs/servers/misc/navidrome/default.nix:45]: A smaller binary is a better binary? I'll just remove them.

**Comment by vs49688** [comment]
Pinging this, now that https://github.com/NixOS/nixpkgs/pull/182665 has hit master.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 182647` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>navidrome</li>
  </ul>
</details>

**Pull request merged**